adapters/repos/db/lsmkv/segment_serialization.go

```diff
@@ -352,32 +362,55 @@ func ParseCollectionNode(r io.Reader) (segmentCollectionNode, error) {
 // As a result calling this method only makes sense if you plan on calling it
 // multiple times. Calling it just once on an uninitialized node does not have
 // major advantages over calling ParseCollectionNode.
-func ParseCollectionNodeInto(in []byte, node *segmentCollectionNode) error {
+func ParseCollectionNodeInto(in io.Reader, node *segmentCollectionNode) error {
```

Comment:
it can be reverted to previous impl, with some slight changes:
- reader as input instead of byte slice (as it is done now)
- additional buffer of size 9 bytes to read "metadata" (tombstones and data sizes), in the same fashion as in `ParseCollectionNode`
- instead of assignments, data can be `pread` directly to node's fields (`node.values[i].tombstone`, `node.values[i].value`, `node.primaryKey`)